api/secrets.proto

```diff
@@ -0,0 +1,133 @@
+syntax = "proto3";
+
+package docker.swarmkit.v1;
+
+import "types.proto";
+import "specs.proto";
+import "objects.proto";
+import "plugin/plugin.proto";
+import "gogoproto/gogo.proto";
+
+// Secrets defines the RPC methods for managing/controlling secrets in a cluster.
+service Secrets {
+	// GetSecret returns a `GetSecretResponse` with a `Secret` with the same
+	// id as `GetSecretRequest.SecretID`
+	// - Returns `NotFound` if the Secret with the given id is not found.
+	// - Returns `InvalidArgument` if the `GetSecretRequest.SecretID` is empty.
+	// - Returns an error if getting fails.
+	rpc GetSecret(GetSecretRequest) returns (GetSecretResponse) {
+		option (docker.protobuf.plugin.tls_authorization) = { roles: "swarm-manager" };
+	}
+
+	// ListSecrets returns a `ListSecretResponse` with a list all `Secret`s being
+	// managed, or all secrets matching any name in `ListSecretsRequest.Names`, any
+	// name prefix in `ListSecretsRequest.NamePrefixes`, any id in
+	// `ListSecretsRequest.SecretIDs`, or any id prefix in `ListSecretsRequest.IDPrefixes`,
+	// filtering by type provided in `ListSecretsRequest.Type`.
+	// - Returns an error if listing fails.
+	rpc ListSecrets(ListSecretsRequest) returns (ListSecretsResponse) {
+		option (docker.protobuf.plugin.tls_authorization) = { roles: "swarm-manager" };
+	}
+	// CreateSecret creates and return a `CreateSecretResponse` with a `Secret`` based
+	// on the provided `CreateSecretRequest.SecretValueSpec`.
+	// - Returns `InvalidArgument` if the `CreateSecretRequest.SecretValueSpec` is malformed.
+	// - Returns `AlreadyExists` if the `CreateSecretRequest.SecretValueSpec.Annotation.Name`
+	//   conflicts with an existing secret's name.
+	// - Returns an error if the creation fails.
+	rpc CreateSecret(CreateSecretRequest) returns (CreateSecretResponse) {
```

Comment:
Should we point out that it also fails if the size of the secret is larger than the `MAX_SIZE`?